authorization/role_assignment_service.go

```diff
@@ -0,0 +1,47 @@
+package authorization
+
+import (
+	"context"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/authorization/models"
+	"github.com/fabric8-services/fabric8-auth/authorization/role"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+)
+
+// RoleAssignmentService defines the contract for managing roles assigments to a resource
+type RoleAssignmentService interface {
+	ListByResource(ctx context.Context, resourceID string) ([]role.IdentityRole, error)
+}
+
+// RoleAssignmentServiceImpl implements the RoleAssignmentService for managing role assignments.
+type RoleAssignmentServiceImpl struct {
+	modelService models.RoleAssignmentModelService
+	db           application.DB
+}
+
+// NewRoleAssignmentService creates a reference to new RoleAssignmentService implementation
+func NewRoleAssignmentService(modelService models.RoleAssignmentModelService, db application.DB) *RoleAssignmentServiceImpl {
+	return &RoleAssignmentServiceImpl{modelService: modelService, db: db}
+}
+
+// ListByResource lists assignments made for a specific resource
+func (r *RoleAssignmentServiceImpl) ListByResource(ctx context.Context, resourceID string) ([]role.IdentityRole, error) {
+
+	var roles []role.IdentityRole
+	var err error
+	err = application.Transactional(r.db, func(appl application.Application) error {
+		err = appl.ResourceRepository().CheckExists(ctx, resourceID)
```

Comment:
Why checking if the resource exists? Want to check if the resource exists before executing an expensive list call?

Reply:
Yes, correct.

Reply:
Good point @sbryzak , having this check helps us with the `NotFoundError` communication. What do you think.. is it worth it?

Reply:
If you wish to return an error if the resource doesn't exist, then yes I think it's worth it.